pkg/controller/kubeletdnat-controller/controller.go

```diff
@@ -275,28 +279,30 @@ func execSave() ([]string, error) {
 	return strings.Split(string(out), "\n"), err
 }
 
-func execRestore(rules []string) error {
-	cmd := exec.Command("iptables-restore", []string{"--noflush", "-v", "-T", "nat"}...)
+func execRestore(ctx context.Context, rules []string) error {
+	cmd := exec.CommandContext(ctx, "iptables-restore", []string{"--noflush", "-v", "-T", "nat"}...)
 
 	stdin, err := cmd.StdinPipe()
 	if err != nil {
 		return err
 	}
-	if _, err := io.WriteString(stdin, strings.Join(rules, "\n")+"\n"); err != nil {
-		return fmt.Errorf("failed to write to iptables-restore stdin: %w", err)
-	}
-	if err := stdin.Close(); err != nil {
-		return fmt.Errorf("failed to close iptables-restore stdin: %w", err)
-	}
+
+	// if input is bigger than OS' pipe buffer size, the write call blocks indefinitely
+	go func() {
+		// ignore errors here, since it would likely lead to either the command to time out or to fail
+		_, _ = io.WriteString(stdin, strings.Join(rules, "\n")+"\n")
+		stdin.Close()
+	}()
 
 	out, err := cmd.CombinedOutput()
-	if err == nil {
-		return nil
-	}
-	if len(out) > 0 {
-		return fmt.Errorf("iptables-restore failed: %w (output: %s)", err, string(out))
+	if err != nil {
+		if len(out) > 0 {
+			return fmt.Errorf("iptables-restore failed: %v (output: %s)", err, string(out))
```

Comment:
Please use `%w` instead of `%v` to wrap errors (this is why [`pre-kubermatic-lint`](https://public-prow.loodse.com/view/gs/prow-dev-public-data/pr-logs/pull/kubermatic_kubermatic/9996/pre-kubermatic-lint/1537054311412928512) failed).

Reply:
Ping @squ94wk, any update on this PR?